pkg/cmd/admin/policy/reconcile_sccs.go

```diff
@@ -0,0 +1,241 @@
+	changedSCCs := []*kapi.SecurityContextConstraints{}
+
+	groups, users := bootstrappolicy.GetBoostrapSCCAccess()
+	bootstrapSCCs := bootstrappolicy.GetBootstrapSecurityContextConstraints(groups, users)
+
+	for i := range bootstrapSCCs {
+		expectedSCC := &bootstrapSCCs[i]
+		actualSCC, err := o.SCCClient.Get(expectedSCC.Name)
+		// if not found it needs to be created
+		if kapierrors.IsNotFound(err) {
+			changedSCCs = append(changedSCCs, expectedSCC)
+			continue
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		// if found then we need to diff to see if it needs updated
+		if updatedSCC, needsUpdating := computeUpdatedSCC(*expectedSCC, *actualSCC); needsUpdating {
+			changedSCCs = append(changedSCCs, updatedSCC)
+		}
+	}
+	return changedSCCs, nil
+}
+
+// ReplaceChangedSCCs persists the changed SCCs.
+func (o *ReconcileSCCOptions) ReplaceChangedSCCs(changedSCCs []*kapi.SecurityContextConstraints) error {
+	for i := range changedSCCs {
+		_, err := o.SCCClient.Get(changedSCCs[i].Name)
+		if err != nil && !kapierrors.IsNotFound(err) {
+			return err
+		}
+
+		if kapierrors.IsNotFound(err) {
+			createdSCC, err := o.SCCClient.Create(changedSCCs[i])
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(o.Out, "securitycontextconstraints/%s\n", createdSCC.Name)
+			continue
+		}
+
+		updatedSCC, err := o.SCCClient.Update(changedSCCs[i])
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(o.Out, "securitycontextconstraints/%s\n", updatedSCC.Name)
+	}
+	return nil
+}
+
+// computeUpdatedSCC compares expected and actual SCCs for changes.  It ignores differences
+// in ObjectMeta.  It will maintain additional grants on the actual SCC by migrating them to
+// the expected SCC in addition to any expected grants.
+func computeUpdatedSCC(expected kapi.SecurityContextConstraints, actual kapi.SecurityContextConstraints) (*kapi.SecurityContextConstraints, bool) {
+	needsUpdate := false
+	// make a copy of the expected scc here so we can swap in things we're going to ignore
+	// (metadata, groups, users).  This allows us to use a semantic deep equal on the
+	// majority of the fields.
+	updated := expected
+	expected.ObjectMeta = actual.ObjectMeta
+	expected.Groups = actual.Groups
+	expected.Users = actual.Users
+
+	if !kapi.Semantic.DeepEqual(expected, actual) {
+		needsUpdate = true
+	}
+
+	// now compare the users and groups
+	if !hasGrants(updated.Groups, actual.Groups) || !hasGrants(updated.Users, actual.Users) {
+		needsUpdate = true
+	}
+
+	// maintain additional grants that the actual objects may have and add any bootstrap grants
+	// that do not exist.
+	updated.Groups = uniqueGrants(updated.Groups, actual.Groups)
+	updated.Users = uniqueGrants(updated.Users, actual.Users)
+	return &updated, needsUpdate
+}
+
+// hasGrants ensures all expected grants are in the actual grants.
+func hasGrants(expectedGrants []string, actualGrants []string) bool {
+	grants := sets.NewString(actualGrants...)
```

Comment:
there are existing `sets.String` functions for checking equality, contains all, etc